File: examples/py/basic_allow.py
from promptsecurity import run_promptsecurity

review = run_promptsecurity(user="Explain CORS and how browsers enforce it.")

if review["action"] == "allow":
    print("allowed, forward to model")
elif review["action"] == "sanitize":
    print("sanitize first:\n", review["sanitized_prompt"])
else:
    print("blocked:", review["reason"])

File: examples/js/basic-allow.ts
import promptsecurity from "promptsecurity"

const review = promptsecurity.scan({
  user: "Summarize how HTTPS handshakes work for a beginner.",
})

if (review.action === "allow") {
  console.log("allowed, forward to LLM")
} else if (review.action === "sanitize") {
  console.log("sanitize first:", review.sanitized_prompt)
} else {
  console.error("blocked:", review.reason)
}
